common/src/chain/transaction/output/mod.rs

```diff
@@ -87,6 +91,28 @@ pub enum TxOutput {
     CreateDelegationId(Destination, PoolId),
     #[codec(index = 6)]
     DelegateStaking(Amount, DelegationId),
+    #[codec(index = 7)]
+    Tokens(TokenOutput),
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode, serde::Serialize)]
+pub enum TokenOutput {
+    #[codec(index = 0)]
+    IssueFungibleToken(Box<TokenIssuance>),
+    // Create certain amount of tokens and add them to circulating supply
+    #[codec(index = 1)]
+    MintTokens(TokenId, Amount, Destination),
+    // Take tokens out of circulation. Not the same as Burn because redemption means that certain amount
+    // of tokens is no longer supported by underlying fiat currency, which can only be done by
+    // reissuance controller.
+    #[codec(index = 2)]
+    RedeemTokens(TokenId, Amount),
```

Comment:
I would call this "RestockTokens". Maybe the natively English speaking dudes here can help with a better term.

Reply:
Maybe @muursh can help. Imo redeem looks good